sample_metadata/parser/generic_metadata_parser.py

```diff
@@ -268,21 +275,22 @@ async def validate_sample_map(self, sample_map: Dict[str, List[Dict[str, Any]]])
                 raise ValueError(f'Unexpected type {type(row)} {row}')
 
         errors = []
-        errors.extend(self.check_files_covered_by_rows(ungrouped_rows))
+        errors.extend(await self.check_files_covered_by_rows(ungrouped_rows))
         if errors:
             raise ValueError(', '.join(errors))
 
-    def check_files_covered_by_rows(
+    async def check_files_covered_by_rows(
         self, rows: List[Dict[str, Any]]
     ) -> List[str]:
         """
         Check that the files in the search_paths are completely covered by the sample_map
         """
-        filenames = []
+        filename_promises = []
         for grp in rows:
             for r in grp if isinstance(grp, list) else [grp]:
-                filenames.extend(r.get(self.reads_column, '').split(','))
+                filename_promises.append(self.get_read_filenames('TEST', r))
```

Comment:
Should this 'TEST' be here?

Reply:
Thank you! I initially thought it wouldn't matter, but there's an easy fix anyway, addressed :)